File: C#/Problems/Problems/Problem3.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Problems
{
    class Problem3 : Problem
    {
        private int[] numbers;
        private int exp;

        public Problem3(int[] numbers, int expected)
        {
            this.numbers = numbers;
            this.exp = expected;
        }

        public bool Calculate()
        {
            int counter = 0;
            HashSet<int> set = new HashSet<int>();
            foreach (int num in numbers)
            {
                if (num == counter + 1)
                {
                    counter++;
                    while (set.Contains(counter + 1))
                    {
                        counter++;
                    }
                }
                else
                {
                    set.Add(num);
                }
            }

            return counter+1 == this.exp;
        }

        
    }
}

File: Javascript/Problem/Problem/js/Problem3.js
function Problem3(list, exp) {
    this.list = list;
    this.exp = exp;
}

Problem3.prototype.Calculate = function () {


    var counter = 0;
    const set = new Set([]);
    this.list.forEach(function (element) {
        if (element == counter + 1) {
            counter++;
            while (set.has(counter + 1)) {
                counter++;
            }
        } else {
            set.add(element);
        }
    });

    return counter+1 == this.exp;
};